authz/rbac_translator.go

```diff
@@ -302,11 +305,20 @@ func (options *auditLoggingOptions) toProtos() (allow *v3rbacpb.RBAC_AuditLoggin
 		if config.Config == nil {
 			return nil, nil, fmt.Errorf("AuditLogger Config field cannot be nil")
 		}
-		customConfig, err := anypb.New(config.Config)
+
+		typedStruct := &v1typepb.TypedStruct{
+			TypeUrl: typedUrlPrefix + config.Name,
+			Value:   config.Config,
+		}
+		if err != nil {
+			return nil, nil, fmt.Errorf("error creating custom config: %v", err)
+		}
+		customConfig, err := anypb.New(typedStruct)
 		if err != nil {
 			return nil, nil, fmt.Errorf("error parsing custom audit logger config: %v", err)
 		}
-		logger := &v3corepb.TypedExtensionConfig{Name: config.Name, TypedConfig: customConfig}
+
+		logger := &v3corepb.TypedExtensionConfig{Name: config.Name /*this doesn't matter*/, TypedConfig: customConfig}
```

Comment:
nit: Maybe elaborate a little more on why this does not matter.

Reply:
I'm just going to remove that comment actually

Reply:
If that is good with you please resolve